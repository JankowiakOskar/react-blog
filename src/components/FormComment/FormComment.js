import React from 'react';
import PropTypes from 'prop-types';

const FormComment = ({ onSubmit, onChange, value }) => (
  <form onSubmit={onSubmit}>
    <input
      type="text"
      placeholder="Type your comment..."
      value={value}
      onChange={onChange}
    />
    <button type="submit">Add comment</button>
  </form>
);

FormComment.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
};

FormComment.defaultProps = {
  value: '',
};

export default FormComment;
