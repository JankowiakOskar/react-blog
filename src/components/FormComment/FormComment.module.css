.form-wrapper {
}

.form-wrapper__title {
  margin: 0 0 2rem 0;
  color: rgb(50, 62, 190);
}

.form {
  width: 100%;
}

.form__input-wrapper {
  margin: 0 0 2.2rem 0;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.form__label {
  margin: 0 0 0.8rem 0;
  font-size: 1.7rem;
  font-weight: bold;
}

.form__input-text {
  padding: 0.8rem 0.5rem;
  border-radius: 0.5rem;
  border: 0.15rem solid #000;
}

.form__text-area {
  resize: none;
  width: 25rem;
  height: 8rem;
}

.form__button {
  padding: 0.8rem;
  background-color: rgb(50, 62, 190);
  color: rgb(255, 255, 255);
  border-radius: 0.5rem;
  font-weight: bold;
  box-shadow: 0rem 0rem 0.2rem 0rem rgba(0, 0, 0, 0.2);
  transition: 0.15s ease;
  border: none;
}

.form__button:hover {
  background-color: rgb(64, 72, 155);
}

@media (min-width: 768px) {
  .form__text-area {
    width: 35rem;
  }
}
