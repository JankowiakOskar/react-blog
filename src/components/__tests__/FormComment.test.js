/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import FormComment from '../FormComment/FormComment';

describe('<FormComment/>', () => {
  let wrapper;
  let testState;
  beforeEach(() => {
    testState = {
      author: '',
      comment: '',
      error: null,
      isDisabled: false,
      isSubmitting: false,
    };

    const mockSetCommentFn = (e) =>
      jest.fn(() => {
        testState.comment = e.target;
      });

    wrapper = shallow(
      <FormComment
        onSubmit={() => {
          testState.isSubmitting = !testState.isSubmitting;
        }}
        setComment={mockSetCommentFn}
        setAuthor={(e) => {
          testState[e.target.name] = e.target.value;
        }}
        author={testState.author}
        isDisabled={testState.isDisabled}
        isSubmitting={testState.isSubmitting}
      />,
    );
  });

  it('should render message during submitting form', () => {
    wrapper.find('form').simulate('submit');
    expect(testState.isSubmitting).toBeTruthy();
  });

  it('should set author name when name has been typed into author input', () => {
    const authorInput = wrapper.find('#author-comment');

    authorInput.simulate('change', {
      target: { name: 'author', value: 'Oskar' },
    });

    expect(testState.author).toEqual('Oskar');
  });

  it('should render error if passed by props', () => {
    const errorMsg = 'Oups we got an error';
    wrapper.setProps({ error: errorMsg });
    expect(wrapper.find('#comment-error').text()).toEqual(`${errorMsg}!`);
  });
});
