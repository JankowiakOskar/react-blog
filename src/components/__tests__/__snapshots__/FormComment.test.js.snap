// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormComment/> renders FormComment component propertly 1`] = `ShallowWrapper {}`;
